### PR TITLE
skip serialize-messages check for well known local messages, #22680

### PR DESCRIPTION
**Pull request by patriknw**
* when serialize-messages=on and allow-java-serialization=off

Refs #22680

### CODE REVIEW

File: akka-actor/src/main/scala/akka/actor/dungeon/Dispatch.scala [hunk 1]
```diff
@@ -157,13 +158,17 @@ private[akka] trait Dispatch { this: ActorCell ⇒
...
     val s = SerializationExtension(system)
     val serializer = s.findSerializerFor(obj)
-    val bytes = serializer.toBinary(obj)
-    serializer match {
-      case ser2: SerializerWithStringManifest ⇒
-        val manifest = ser2.manifest(obj)
-        s.deserialize(bytes, serializer.identifier, manifest).get
-      case _ ⇒
-        s.deserialize(bytes, obj.getClass).get
+    if (serializer.isInstanceOf[DisabledJavaSerializer] && !s.shouldWarnAboutJavaSerializer(obj.getClass, serializer))
```
Comment (ktoso) [inline akka-actor/src/main/scala/akka/actor/dungeon/Dispatch.scala]: Ah! Smart. Glad that we've explicitly set it to this special one 😃

**Review by ktoso — APPROVED** [review]
LGTM, very nice

**Comment by patriknw** [comment]
I fixed some part of the failing test, but there is still one thing that I don't fully follow (that weird edge case):

```
[info] Disabling java serialization
[WARN] [SECURITY][04/05/2017 15:44:41.813] [pool-112-thread-1-ScalaTest-running-SerializationSetupSpec] [DisabledJavaSerializer(akka://addedJavaSerializationSystem)] Outgoing message attempted to use Java Serialization even though `akka.actor.allow-java-serialization = off` was set! Message type was: [class java.util.BitSet]
[info] - should throw if passed system to JavaSerializer has allow-java-serialization = off (3 milliseconds)
[info] - should have replaced java serializer *** FAILED *** (16 milliseconds)
[info]   java.lang.AssertionError: assertion failed: received unexpected message {}
[info]   at scala.Predef$.assert(Predef.scala:170)
[info]   at akka.testkit.TestKitBase$class.expectNoMsg_internal(TestKit.scala:631)
[info]   at akka.testkit.TestKitBase$class.expectNoMsg(TestKit.scala:627)
[info]   at akka.testkit.TestKit.expectNoMsg(TestKit.scala:829)
[info]   at akka.serialization.SerializationSetupSpec$$anonfun$2$$anonfun$apply$mcV$sp$4.apply(SerializationSetupSpec.scala:142)
```

**Comment by patriknw** [comment]
found it, sorry for the noise

**Review by ktoso — APPROVED** [review]
LGTM again 👍

**Review by johanandren — APPROVED** [review]
LGTM, nice!

**Comment by patriknw** [comment]
Changed my mind regarding backporting this. It's not trivial, seems like release-2.4 and master have diverged, and since this is not a production critical feature I don't think we should spend time on backporting it.
